bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/HSBType.java

```diff
@@ -257,6 +295,57 @@ public boolean equals(Object obj) {
         return new PercentType[] { red, green, blue };
     }
 
+    // Gamma compression (sRGB) for a single component, in the 0.0 - 1.0 range
+    private static float gammaCompress(float c) {
+        if (c < 0.0f) {
+            c = 0.0f;
+        } else if (c > 1.0f) {
+            c = 1.0f;
+        }
+
+        return c <= 0.0031308f ? 19.92f * c : (1.0f + 0.055f) * (float) Math.pow(c, 1.0f / 2.4f) - 0.055f;
+    }
+
+    // Gamma decompression (sRGB) for a single component, in the 0.0 - 1.0 range
+    private static float gammaDecompress(float c) {
+        if (c < 0.0f) {
+            c = 0.0f;
+        } else if (c > 1.0f) {
+            c = 1.0f;
+        }
+
+        return c <= 0.04045f ? c / 19.92f : (float) Math.pow((c + 0.055f) / (1.0f + 0.055f), 2.4f);
+    }
+
+    /**
+     * Returns the xyY values representing this object's color in CIE XY color model.
+     * Conversion from sRGB to CIE XY using D65 reference white
+     * xy pair contains color information
+     * Y represents relative luminance
+     *
+     * @param HSBType color object
+     * @return PercentType[x, y, Y] values in the CIE XY color model
+     */
+    public PercentType[] toXY() {
```

Comment:
The `fromXY(...)` method above takes `float` arguments, while `toXY(...)` here returns a `PrecentType` array. Is this asymmetry intended?

Reply:
I implemeted this trying to be coherent with the existing fromRGB / toRGB couple of methods already in HSBtype (fromRGB takes ints, while toRGB returns Percentypes). Other than that, there is no particular reason. I would prefer to use only floats in the conversions, but it is like this just for consistency